scene/main/viewport.cpp

```diff
@@ -1618,7 +1618,10 @@ void Viewport::_gui_show_tooltip() {
 	gui.tooltip_popup->set_position(r.position);
 	gui.tooltip_popup->set_size(r.size);
 
-	gui.tooltip_popup->show();
+	if (window->has_focus()) {
```

Comment:
It seems a bit wasteful to do all those calculations to finally not show the tooltip... maybe the check could be an early return at the start instead? Or will the instantiated `gui.tooltip_popup` be useful later on even if it wasn't shown?

Reply:
Yeah, I thought it was pretty wasteful too, but every time I tried to move the definition of the window up higher in the function, Godot crashed. I'm not really sure what I'm missing...